File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maximum  = min(nums)
        sum = 0
        for i in range(len(nums)):
            sum += nums[i]
            maximum = max(sum, maximum)
            if sum < 0:
                sum  = 0
        return maximum

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = 0;
        int max = 0;
        for(int num:nums){
            max =  max<num?max:num;
        }
        for(int num:nums){
            sum+=num;
            max = max>sum?max:sum;
            if(sum<0){
                sum=0;
            }
        }
        return max;
    }
}